### PR TITLE
[enhancement](profile) get real-time query profile

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
successor of https://github.com/apache/doris/pull/33331

step 2 of https://github.com/apache/doris/issues/33744

When getting profile of a running query, we will fetch & update profile data in a synchronous manner.

Actually, this pr just finished real-time profile fetching on FE, we need more work on BE. For example, profile should be updated during each time pipeline task is executed instead of only on close stage.

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run compile

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 1]
```diff
@@ -4083,6 +4087,54 @@ public List<QueryStatisticsItem.FragmentInstanceInfo> getFragmentInstanceInfos()
...
+                    continue;
+                }
+                LOG.info("Get real-time exec status succeed, query {} backend {}",
+                            DebugUtil.printId(queryId), address.toString());
+                QeProcessorImpl.INSTANCE.reportExecStatus(resp.getReportExecStatusParams(), address);
+            } catch (Exception e) {
+                LOG.warn("Got exception when getRealtimeExecStatus, query {} backend {}",
+                        DebugUtil.printId(queryId), address.toString(), e);
+            }
+        }
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:4135]: should return object

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 2]
```diff
@@ -4083,6 +4087,54 @@ public List<QueryStatisticsItem.FragmentInstanceInfo> getFragmentInstanceInfos()
...
+                backendAddresses.add(new TNetworkAddress(backend.getHost(), backend.getBePort()));
+            }
+        } else {
+            for (Long backendId : this.beToExecStates.keySet()) {
+                Backend backend = idToBackend.get(backendId);
+                backendAddresses.add(new TNetworkAddress(backend.getHost(), backend.getBePort()));
+            }
+        }
+
+        for (TNetworkAddress address : backendAddresses) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:4105]: If there are many BEs, call one by one is very slow.

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java (hunk 1 above)
Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:4135]: signature of this function is void

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java (hunk 2 above)
Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:4105]: I will change this to parallel

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 3]
```diff
@@ -4083,6 +4087,54 @@ public List<QueryStatisticsItem.FragmentInstanceInfo> getFragmentInstanceInfos()
...
+            try {
+                Client client = ClientPool.backendPool.borrowObject(address);
+                TGetRealtimeExecStatusRequest req = new TGetRealtimeExecStatusRequest();
+                req.setId(this.queryId);
+                TGetRealtimeExecStatusResponse resp = client.getRealtimeExecStatus(req);
```
[the remark this answers is not in the document]
Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:4110]: No, i will use future and set timeout

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
image
fail of FEUT should be ignored

**Comment by zhiqiang-hhhh** [comment]
run feut

**Comment by zhiqiang-hhhh** [comment]
run feut

**Comment by zhiqiang-hhhh** [comment]
run compile

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
image
http://43.132.222.7:8111/test/8774097394913849745?currentProjectId=Doris_DorisRegression&expandTestHistoryChartSection=true

P0 Failed has nothing to do with this PR

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**